www/pages/routes-and-stops/routes-and-stops-controller.js

```diff
@@ -111,22 +126,53 @@ angular.module('pvta.controllers').controller('RoutesAndStopsController', functi
       }
     });
   };
-  function getFavorites () {
+
+  function getFavoriteRoutes (routes) {
     localforage.getItem('favoriteRoutes', function (err, value) {
       $scope.favoriteRoutes = value;
+      $scope.routes = stripDetails(routes);
       redraw();
     });
+  };
+
+  function getFavoriteStops (stops){
     localforage.getItem('favoriteStops', function (err, value) {
       $scope.favoriteStops = value;
+      $scope.stops = prepareStops(stops);
       redraw();
     });
-  }
+  };
+
+  $scope.toggleStopHeart = function (stop) {
+    FavoriteStops.contains(stop.StopId, function (bool) {
+      if (bool) {
+        FavoriteStops.remove(stop);
+      }
+      else {
+        FavoriteStops.push(stop);
+      }
+      $scope.$apply();
+    });
+  };
+
+  $scope.toggleRouteHeart = function(route){
+    FavoriteRoutes.contains(route, function(bool){
+      if(bool) {
+        FavoriteRoutes.remove(route);
+      }
+      else {
+        FavoriteRoutes.push(route);
+      }
+      $scope.$apply();
+    });
+  };
 
   function redraw () {
     $scope.display($scope.currentDisplay);
   }
-  getRoutesAndStops();
+
   $scope.$on('$ionicView.enter', function () {
-    getFavorites();
+    getRoutesAndStops();
+    $scope.display($scope.currentDisplay);
```

Comment:
We have a [function](https://github.com/umts/pvta-multiplatform/pull/268/files?diff=unified#diff-365f07089626b424259d271203b2ff1eR170) (`redraw`) that does this for us...call it.